src/Misc/Primes.jl

```diff
@@ -116,6 +116,8 @@ struct PrimesSet{T}
       end
       p = next_prime(p, false)
     end
+    mod = abs(mod)
+    val = abs(val)
```

Comment:
```suggestion
    val = mod(val, mod)
    val = val < 0 ? val + mod : val
```
Does this make sense? I think we allow the restriction to be "primes which are -12 modulo 5".